testdata/p4_16_samples/parser-non-field-lhs-bmv2.p4

```diff
@@ -0,0 +1,73 @@
+// Test that non-field LHS works in parser on BMv2.
+
+#include <core.p4>
+#include <v1model.p4>
+
+header test_h {
+    bit<8> field;
+}
+
+struct header_t {
+    test_h[4] hs;
+}
+
+struct metadata_t {
+    bit<2> hs_next_index;
+}
+
+parser TestParser(packet_in pkt,
+                  out header_t hdr,
+                  inout metadata_t meta,
+                  inout standard_metadata_t std_meta) {
+    state start {
+        hdr.hs[meta.hs_next_index].setValid();
+        hdr.hs[meta.hs_next_index] = {8w0};
```

Comment:
The initial value of meta.hs_next_index is unspecified in this program.  Is that what you intended?  If not, it would be better to explicitly initialize meta.hs_next_index before its first use, e.g. by assigning it 0.

Reply:
This is a good solution for this PR. In the general case, ParserUnroll can't generate an exception for such examples. So, I'll try to fix it.

Reply:
I am actually a bit surprised that there is no warning from p4c that `meta.hs_next_index` is used without being initialized.

Reply:
Thanks for catching this! I've added the initialization as you suggested.